### DISCUSSION TITLE
Translate strings from my backend

### DISCUSSION THREAD

**Discussion by giofid** [SwiftGen/SwiftGen#908] [Q&A] [answered]
I'm trying a way to use SwiftGen to translate strings that I obtain from my backend. Something like

```
let myBeStruct = ... // a struct from my backend API call
L10n.MyTable.tr(myBeStruct.myString)
```

MyTable is the name of `MyTable.strings` file containing 

> "myString" = "the translate of myString";

Do I have to create my template, or there is already a way to achieve my goal?

**Comment by djbe** [comment, marked as answer]
I'm assuming you want to be able to use the `tr` function which is `private` in the default generated code, like here: 
https://github.com/SwiftGen/SwiftGen/blob/stable/Sources/TestUtils/Fixtures/Generated/Strings/structured-swift5/localizable.swift#L104-L107

You could just as well create your own function that does the same thing. You can put this in a swift file somewhere, it's actually unrelated to SwiftGen.
```swift
extension L10n {
  static func translateApiString(_ table: String, _ key: String) -> String {
    let format = BundleToken.bundle.localizedString(forKey: key, value: nil, table: table)
    return String(format: format, locale: Locale.current)
  }
}

// or for a specific table
extension L10n.MyTable {
  static func translateApiString(_ key: String) -> String {
    ...
  }
}
```

A couple of things to note though:
- This is *UNSAFE*! It could stop working at any point if your backend returns a key that your app doesn't know.
- What if the backend returns a key only supported by newer app versions? How will you handle this correctly for older versions?
- Should this support arguments? I've left this out of the example above.
- As mentioned above, this seems unrelated to SwiftGen. IMO the API should return translated strings.
